pkg/users/account-details.js

```diff
@@ -355,3 +347,126 @@
         </Page>
     );
 }
+
+export const AccountGroupsSelect = ({ name, groups, setError }) => {
+    const [isOpenGroup, setIsOpenGroup] = useState(false);
+    const [selected, setSelected] = useState();
+    const [primaryGroupName, setPrimaryGroupName] = useState();
+    const [history, setHistory] = useState([]);
+    const previousValue = useRef(null);
+
+    useEffect(() => {
+        const usedGroups = groups.filter(group => group.userlist.includes(name));
+        const primaryGroup = groups.find(group => group.userlistPrimary.includes(name));
+        const _primaryGroupName = primaryGroup ? primaryGroup.name : undefined;
+        const _selected = usedGroups.map(group => group.name);
+        if (primaryGroup)
+            _selected.push(_primaryGroupName);
+
+        previousValue.current = _selected;
+        setSelected(_selected);
+        setPrimaryGroupName(_primaryGroupName);
+    }, [groups, setSelected, name, previousValue]);
+
+    const undoGroupChanges = () => {
+        const undoItem = history[history.length - 1];
+        if (undoItem.type === 'added') {
+            removeGroup(undoItem.name, true).then(() => setHistory(history.slice(0, -1)));
+        } else if (undoItem.type === 'removed') {
+            addGroup(undoItem.name, true).then(() => setHistory(history.slice(0, -1)));
+        }
+    };
+
+    const removeGroup = (group, isUndo) => {
+        if (!isUndo)
+            setHistory([...history, { type: 'removed', name: group }]);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17996-20221212-195940-2c392d36-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)